store/etcd/types_transformation.go

```diff
@@ -0,0 +1,182 @@
+package etcd
+
+import (
+	"fmt"
+
+	"github.com/heetch/regula"
+	"github.com/heetch/regula/rule"
+	pb "github.com/heetch/regula/store/etcd/proto"
+)
+
+// toProtobufRuleset creates a protobuf Ruleset from a regula.Ruleset.
+func toProtobufRuleset(rs *regula.Ruleset) *pb.Ruleset {
+	pbrs := &pb.Ruleset{
+		Type: rs.Type,
+	}
+
+	for _, r := range rs.Rules {
+		pbr := &pb.Rule{
+			Expr:   toProtobufExpr(r.Expr),
+			Result: toProtobufValue(r.Result),
+		}
+		pbrs.Rules = append(pbrs.Rules, pbr)
+	}
+
+	return pbrs
+}
+
+// toProtobufExpr creates a protobuf Expr from a rule.Expr.
+func toProtobufExpr(expr rule.Expr) *pb.Expr {
+	switch e := expr.(type) {
+	case *rule.Value:
+		v := &pb.Value{
+			Type: e.Type,
+			Kind: e.Kind,
+			Data: e.Data,
+		}
+
+		return &pb.Expr{
+			Expr: &pb.Expr_Value{Value: v},
+		}
+	case *rule.Param:
+		p := &pb.Param{
+			Kind: e.Kind,
+			Type: e.Type,
+			Name: e.Name,
+		}
+
+		return &pb.Expr{
+			Expr: &pb.Expr_Param{Param: p},
+		}
+	}
+
+	var (
+		ope rule.Operander
+		ok  bool
+	)
+	if ope, ok = expr.(rule.Operander); !ok {
+		// there is something very weird, a rule.Expr which is not a rule.Value nor a rule.Param nor a rule.Operander
+		// let's panic...
+		panic(fmt.Sprintf("cannot create a pb.Expr - unexpected type: %T", expr))
+	}
+
+	o := &pb.Operator{
+		Kind: expr.Contract().OpCode,
+	}
+	for _, op := range ope.Operands() {
```

Comment:
Same here, it's normally an easy win in terms of memory and performance to simply preallocate the slice when you know the size.